### PR TITLE
[rfw] Add FractionallySizedBox to RFW core widgets.

### PR DESCRIPTION
**Pull request by shuoch-g**
issue 90218

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by shuoch-g** [comment]
just try to get started on rfw package.

File: packages/rfw/lib/flutter/core_widgets.dart [hunk 1]
```diff
@@ -344,6 +344,15 @@ Map<String, LocalWidgetBuilder> get _coreWidgetsDefinitions => <String, LocalWid
     );
   },
 
+  'FractionallySizedBox': (BuildContext context, DataSource source) {
+    return FractionallySizedBox(
+      alignment: (ArgumentDecoders.alignment(source, ['alignment']) ?? Alignment.center).resolve(Directionality.of(context)),
```
Comment (Hixie) [inline packages/rfw/lib/flutter/core_widgets.dart:349]: no need to resolve it, FractionallySizedBox accepts AlignmentGeometry

**Comment by Hixie** [comment]
Change looks good modulo a minor comment.

You'll need to update the CHANGELOG and pubspec.yaml as per the checklist in the PR description.

File: packages/rfw/lib/flutter/core_widgets.dart (hunk 1 above)
Comment (shuoch-g) [reply packages/rfw/lib/flutter/core_widgets.dart]: thanks, updated in a following commit.

**Comment by shuoch-g** [comment]
I got this error on the changelog format:

>  When bumping the version for release, the NEXT section should be incorporated into the new version's release notes.

I guess I need to wait for https://github.com/flutter/packages/pull/474/files#diff-e2c8015a101e034b67a0a3d2834411b5e09885362b9f33f125946de5c3fc53be to got in first since that PR solves the `Fix `unnecessary_import` lint errors.` problem ;)

**Comment by shuoch-g** [comment]
synced and updated, PTAL, thanks !

**Review by Hixie — APPROVED** [review]
LGTM

**Comment by Hixie** [comment]
I just noticed we should also update the docs in core_widgets.dart

**Comment by Hixie** [comment]
https://github.com/flutter/packages/pull/480

**Comment by shuoch-g** [comment]
> I just noticed we should also update the docs in core_widgets.dart

sure, I will keep in mind in following CLs.

I plan to add more from the data we have in http://shortn/_RVaOLeJnWa

**Pull request merged**